pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/BaseRecordExtractor.java

```diff
@@ -186,6 +187,9 @@ protected Object convertSingleValue(Object value) {
     if (value instanceof Number || value instanceof byte[]) {
       return value;
     }
+    if (value instanceof GenericFixed) {
```

Comment:
Is it possible to override this function from `AvroRecordExtractor`?

This seems to be the avro specific logic and we can probably just add it to Avro related record extractor.

Reply:
Done.